tests/Microsoft.Identity.Web.Test.Common/Mocks/MockHttpClientFactory.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Runtime.InteropServices;
+using Microsoft.Identity.Client;
+using Xunit;
+
+namespace Microsoft.Identity.Web.Test.Common.Mocks
+{
+    public class MockHttpClientFactory : IMsalHttpClientFactory, IDisposable
```

Comment:
This is a variation of the HTTP mocking we do in MSAL. It uses public APIs only. It will avoid writing expensive integration tests.

Reply:
Mock* objects do not need to be reviewed, they are mostly copied from MSAL.